### DISCUSSION TITLE
Dev vs. Production Site

### DISCUSSION THREAD

**Discussion by fmanaa** [swellstores/community#77] [Help] [answered]
I am on the community version... When I make changes to the theme and publish from the GUI (Storefront -> Theme -> Edit Theme -> Publish), the changes only affect the https://XYZ--dev.swell.store/ version and not the https://XYZ.swell.store/ or the https://XYZ.apps.swellstores.com/. How do I propagate the change all the way through?

**Comment by jesseswell** [comment, marked as answer]
Hi @fmanaa ! 

The production site should also get updated when publishing via the storefront editor. I checked with my test store under the community plan, and the production site updates whenever I publish changes using the storefront editor. What's your store link? And are you on the latest version of the Origin theme?

**Reply by fmanaa** [reply]
Very odd, I was on Origin v2.0.8. I clicked it and it asked me to upgrade to v2.0.7!!! Once I did that, I see the latest changes on my production site again. Not sure if 2.0.8 was a bad version that was rolled back. Anyhow, thanks!

**Comment by FeMaffezzolli** [comment]
I'm facing the same problem, my dev environment is getting updated when I publish updates, but not the production one.
On the editor I see "Origin V.2.0.10".

**Comment by jesseswell** [comment]
~~@FeMaffezzolli, when you said "publish updates," do you mean making changes to your store using the storefront editor? Your account is on the Trial plan, and you still need to publish your storefront.~~

I just saw your main store under the community plan. Have you checked in incognito mode? It might be cache related.

**Reply by FeMaffezzolli** [reply]
Hey @jesseswell , thanks for your reply.

Yes, correct, I meant "publish updates". That's intriguing that I'm on trial. I'm pretty sure I've added my credit card info yesterday, selected the basic plan and went live. I did it all yesterday successfully.

I'll check that again anyway.

Oh, maybe there's one detail I omitted. I have two stores under my account. The one I'm really working on is the 'iron-shape'.

**Reply by jesseswell** [reply]
@FeMaffezzolli, I just saw you are in the community plan. iron-shape.com, from a separate discussion. I see the changes, though. Are any specific changes not reflected in the live, production version of your store?

**Reply by FeMaffezzolli** [reply]
@jesseswell , yes, there are differences, I'll try to show one here with print screen examples.

For example, on the editor I've added a new collection section, called "Whey Protein".

![image](https://user-images.githubusercontent.com/26548010/214050705-15da5593-f080-4b3c-b67e-4b375d74291f.png)

When I check the preview env, it's there:

![image](https://user-images.githubusercontent.com/26548010/214050819-577b4a9a-a2fd-49e1-80d7-c429cc8b1ed7.png)

But when I  go to live env, even using incognito mode, the section is not there:

<img width="1509" alt="image" src="https://user-images.githubusercontent.com/26548010/214050912-36122710-bde3-4a1b-926f-f5532c61b592.png">

It's just one example, but I'm pretty sure it's happening for every change that I made after I went live for the first time.

**Reply by jesseswell** [reply]
@FeMaffezzolli I tried to access your storefront to investigate, and on the categories page, I see the changes you mentioned, which are also in the dev version of your store.

<img width="1429" alt="image" src="https://user-images.githubusercontent.com/87753957/214053116-9a0c31d0-7f6a-4a9c-8981-c2925a5f3199.png">

I'm using Chrome 108.0.5359.98. Have you checked with a different device?

**Reply by FeMaffezzolli** [reply]
@jesseswell , well that's weird. I've tried on my mobile on Safari and I can't yet see the whey section.

![IMG_09CF99AAAE34-1](https://user-images.githubusercontent.com/26548010/214056534-df50d008-5f62-4ce1-a59d-e07497098dd7.jpeg)

I also checked if accessing the swell generated url versus the custom domain would make any difference, but no. in both cases I still don't see it. My chrome version is 109.0.5414.87

- Auto generated URL

<img width="1082" alt="image" src="https://user-images.githubusercontent.com/26548010/214058400-b9258bee-fae9-487c-bbe6-a91ee2f11497.png">

- Custom domain

![image](https://user-images.githubusercontent.com/26548010/214058524-08b8346e-b79a-4ce0-b86b-1b7d00a95061.png)

To give you more context, I'm using cloudflare DNS services, but I have turned on the dev mode, so we shouldnt expect any kind of cache on their end

![image](https://user-images.githubusercontent.com/26548010/214058710-d56cee00-6334-4887-b5e1-dabdb13e0b66.png)

I've no more ideas to try. Any suggestion from this point?

**Reply by FeMaffezzolli** [reply]
@jesseswell , considering that I have used 4 or 5 different devices (some of them accessing the website for first time) and I got the same result, I think we could remove the hypothesis of device cache, right? Maybe we would be facing a cdn or edge network caching. Is it  plausible?

**Reply by logeshswell** [reply]
@FeMaffezzolli The publishing issue is fixed in your storefront now. Can you check and confirm please?
<img width="1507" alt="Screenshot 2023-06-20 at 11 42 20 PM" src="https://github.com/swellstores/community/assets/100766131/5758b76c-3e8c-40f4-972c-fd94ebaa0e12">

**Reply by Strongeyed** [reply]
Hi @logeshswell I still can't publish from my backend (store editor) is it possible to see why I can't publish? Otherwise, if I make changes in the editor, can I do a manual $git push to my remote repository on Swell to apply the changes?

**Reply by logeshswell** [reply]
@Strongeyed There was a configuration missing in your store as well. I just recovered and updated it. 
You can either do git push or use swell editor to publish your storefront.
